firmware/src/drivers/gpio.h

```diff
@@ -0,0 +1,73 @@
+//
+// Created by Luca on 24/01/2023.
+//
+
+#pragma once
+
+#include "target.h"
+
+namespace drivers {
+
+class OutputPin {
+ public:
+  /** Constructor
+   *
+   * @param port Pointer to the port
+   * @param pin Pin number
+   */
+  OutputPin(GPIO_TypeDef* port, uint16_t pin) : m_port(port), m_pin(1U << pin) {
+    // TODO actually init the pin here
+  }
+
+  /** Set output high
+   *
+   */
+  void SetHigh() { m_port->BSRR = m_pin; }
+
+  /** Set output low
+   *
+   */
+  void SetLow() { m_port->BSRR = (m_pin << 16U); }
```

Comment:
any description on what these things do would be useful

Reply:
It says right there, set the output pin low